tensorflow/lite/kernels/unpack_test.cc

```diff
@@ -67,192 +66,97 @@ class UnpackOpModel : public SingleOpModel {
   std::vector<int> outputs_;
 };
 
-// float32 tests.
-TEST(UnpackOpTest, FloatThreeOutputs) {
-  UnpackOpModel<float> model({TensorType_FLOAT32, {3, 2}}, 0);
-  model.SetInput({1, 2, 3, 4, 5, 6});
-  model.Invoke();
+template <typename T>
+void Check(int axis, const std::initializer_list<int>& input_shape,
+           const std::initializer_list<T>& input_data,
+           const std::vector<std::vector<int>>& exp_output_shape,
+           const std::vector<std::vector<T>>& exp_output_data,
+           const TensorType& type = TensorType_FLOAT32) {
```

Comment:
nit: for integer, it's actually fine not to do reference

Reply:
follow the other existing design changes, need to change the reference?

Reply:
up to you, I'm just saying it's not required for integer

Reply:
okay